packages/design-system-web/src/components/geov-yasgui-map-circles/geov-yasgui-map-circles.tsx

```diff
@@ -0,0 +1,214 @@
+import { Component, Host, h, Prop, Element, State } from '@stencil/core';
+import { isNode } from '../../lib/isNode';
+import type { Parser } from '@triply/yasr';
+import { importMapLibre } from '../../lib/importMapLibre';
+import { DataDrivenPropertyValueSpecification, LngLatBoundsLike } from 'maplibre-gl';
+
+function createGeoJSON(data: Parser.Binding[]) {
+  const labelIndices = [...new Set(data.map(ele => ele['type']?.value || 'none'))];
+  return {
+    type: 'FeatureCollection',
+    features: data.map(ele => {
+      return {
+        type: 'Feature',
+        geometry: {
+          type: 'Point',
+          coordinates: [parseFloat(ele['long'].value), parseFloat(ele['lat'].value)],
+        },
+        //set the properties to the .value of the ele
+        properties: {
+          label: ele['label']?.value,
+          radius: Math.round((parseFloat(ele['radius']?.value) || 0) * 10) / 10,
+          number: parseInt(ele['number']?.value),
+          type: ele['type']?.value || 'none',
+          typeindex: labelIndices.indexOf(ele['type']?.value || 'none'),
+          link: ele['link']?.value,
+        },
+      };
+    }),
+  };
+}
+/**
+ * This component is used by Yasgui as a plugin. It consumes the data from Yasgui.
+ *
+ * The result is displayed as circles of different sizes on a map.
+ */
+@Component({
+  tag: 'geov-yasgui-map-circles',
+  styleUrl: 'geov-yasgui-map-circles.css',
+  shadow: false,
+})
+export class GeovYasguiMapCircles {
+  @Element() el: HTMLElement;
+
+  @State() loading: boolean;
+
+  @Prop() data: Parser.Binding[] = [
+    {
+      radius: { value: '80.2345', type: 'literal' },
+      numbers: { value: '0', type: 'literal' },
+      types: { value: 'default', type: 'literal' },
+      long: { value: '8.2318', type: 'literal' },
+      lat: { value: '46.7985', type: 'literal' },
+      labels: { value: 'default', type: 'literal' },
+      link: { value: 'default', type: 'literal' },
+    },
+  ];
+
+  @Prop() colorScale: string[] = ['#a6cee3', '#1f78b4', '#b2df8a', '#33a02c']; // ColorBrewer2 qualitative 4-class Paired (colorblind safe)
+
+  async componentDidLoad() {
+    // If we are in a browser
+    if (!isNode()) {
+      this.loading = true;
+
+      const invalidPoints = this.data.filter(d => !d['long'] || !d['lat']);
```

Comment:
filters only on empty long/lat variables
you can enter a string of characters ('adcz') and it would act as an empty long/lat